ykrt/src/compile/j2/x64/x64hir_to_asm.rs

```diff
@@ -1453,23 +1453,20 @@ impl HirToAsmBackend for X64HirToAsm<'_> {
         And { tyidx: _, lhs, rhs }: &And,
     ) -> Result<(), CompilationError> {
         let bitw = b.inst_bitw(self.m, *lhs);
-        let (imm, mut in_fill) = (
-            self.zero_ext_op_for_imm32(b, bitw, *rhs),
-            RegCnstrFill::Zeroed,
-        );
-        if bitw == 32 || bitw == 64 {
-            // We can relax sign / zero fill for values that are exactly 32/64 bit.
-            in_fill = RegCnstrFill::Undefined;
-        }
+        let out_fill = if bitw == 32 || bitw == 64 {
+            RegCnstrFill::Zeroed
+        } else {
+            RegCnstrFill::Undefined
+        };
 
-        if let Some(imm) = imm {
+        if let Some(imm) = self.zero_ext_op_for_imm32(b, bitw, *rhs) {
```

Comment:
I'm a little confused here, the docs say for a 32-bit immediate and 64 bit register, the 32-bit value is sign-extended to 64-bit. https://www.felixcloutier.com/x86/and

However, in this case, we check against the zero extended version. Why is that?

Reply:
Reading a little more into this https://github.com/ykjit/yk/blob/8f4d9680caee83c7875dd14e87a09ce070df90ec/ykrt/src/compile/j2/x64/x64hir_to_asm.rs#L142

It seems this is safe?